src/Survey/ItechArt.Survey.Foundation/Counters/DatabaseCounterService.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using ItechArt.Repositories.Abstractions;
+using ItechArt.Survey.DomainModel;
+using ItechArt.Survey.Foundation.Counters.Abstractions;
+
+namespace ItechArt.Survey.Foundation.Counters;
+
+public class DatabaseCounterService : ICounterService
+{
+    private readonly IUnitOfWork _unitOfWork;
+
+
+    public DatabaseCounterService(IUnitOfWork unitOfWork)
+    {
+        _unitOfWork = unitOfWork;
+    }
+
+
+    public async Task<Counter> GetCounterAsync()
+    {
+        var counter = (await _unitOfWork.GetRepository<Counter>().GetAllAsync()).FirstOrDefault();
```

Comment:
(move repository to separate variable)
(move get all to separate variable)
add first or default method to repository interface (and implement it)